lib/main.dart

```diff
@@ -302,32 +271,75 @@ class SplashState extends ConsumerState<Splash> with WidgetsBindingObserver {
       FlutterNativeSplash.remove();
 
       if (session.isLoggedOut) {
-        await _goToIntroScreen();
+        _goToIntroScreen();
         return;
       }
 
       Navigator.of(context).pushReplacementNamed('/home');
     } catch (e, stack) {
       dev.log(e.toString(), error: e, stackTrace: stack);
       FlutterNativeSplash.remove();
-      await _goToIntroScreen();
+      _goToIntroScreen();
+    }
+  }
+
+  Future _checkDeviceSecurity() async {
+    if (Platform.isIOS == false && Platform.isAndroid == false) {
+      return;
+    }
+
+    final preferences = await HivePreferencesDatasource.getInstance();
+    if (await SecurityManager().isDeviceJailbroken() == false &&
+        await SecurityManager().isDeviceDeveloperMode() == false) {
+      // user will see a popup next time his device is jailbroken or in dev mode
+      await preferences.setHasNotSeenRootWarning();
+      return;
+    }
+
+    final localizations = AppLocalizations.of(context)!;
+
+    // User never saw the error saying his device is unsafe, we will let him know that there is a mistake via a popup
+    // Next time he will launch the app he will only see a snack bar
+    if (preferences.getHasSeenRootWarning() == false) {
+      final language = ref.read(
+        LanguageProviders.selectedLanguage,
+      );
+      AppDialogs.showInfoDialog(
+        context,
+        ref,
+        CaseChange.toUpperCase(localizations.warning, language.getLocaleString()),
+        localizations.rootWarning,
+        buttonLabel: AppLocalizations.of(context)!.iUnderstandTheRisks.toUpperCase(),
+        onPressed: () async {
+          await preferences.setHasSeenRootWarning();
+        },
+      );
+    } else {
+      final theme = ref.watch(ThemeProviders.selectedTheme);
+      UIUtil.showSnackbar(
+        localizations.rootWarning,
+        context,
+        ref,
+        theme.text!,
+        theme.snackBarShadow!,
+        duration: const Duration(seconds: 10),
+      );
     }
   }
 
-  Future<void> _goToIntroScreen() async {
-    await Navigator.of(context).pushReplacementNamed('/intro_welcome');
+  void _goToIntroScreen() {
+    Navigator.of(context).pushReplacementNamed('/intro_welcome');
```

Comment:
synchronous method now because the await will end when the page "intro_welcome" has been pushed in the navigation stack and is not in the top anymore. 
I add some code after the call of _goToIntroScreen, that's why I needed to change that.